src/utils/AddressUtils.ts

```diff
@@ -55,3 +55,12 @@ export function toAddress(bytes32: string): string {
   const rawAddress = utils.hexZeroPad(utils.hexStripZeros(bytes32), 20);
   return utils.getAddress(rawAddress);
 }
+
+// Checks if an input address is a 32-byte address or not.
+export function isAddressBytes32(address: string): boolean {
```

Comment:
It might be good to do something like what viem does here and define a type `Address` and enforce them to be `0x${string}`, maybe even of certain lengths if we can do that in TypeScript.